src/main/java/com/faforever/client/game/CreateGameController.java

```diff
@@ -407,26 +409,45 @@ private void onGenerateMap() {
   }
 
   public void onCreateButtonClicked() {
-    MapBean selectedMap = mapListView.getSelectionModel().getSelectedItem();
     onCloseButtonClicked();
+    hostGameAfterMapUpdate();
+  }
+
+  private void hostGameAfterMapUpdate() {
+    MapBean selectedMap = mapListView.getSelectionModel().getSelectedItem();
     if (mapService.isOfficialMap(selectedMap)) {
-      hostGame(selectedMap);
+      hostGameAfterModUpdate(selectedMap);
     } else {
       mapService.updateLatestVersionIfNecessary(selectedMap)
           .exceptionally(throwable -> {
-            log.error("error when updating the map", throwable);
-            hostGame(selectedMap);
+            log.error("Error when updating the map", throwable);
+            hostGameAfterModUpdate(selectedMap);
             return null;
           })
-          .thenAccept(this::hostGame);
+          .thenAccept(this::hostGameAfterModUpdate);
     }
   }
 
-  private void hostGame(MapBean map) {
-    Set<String> mods = modManagerController.apply().stream()
+  private void hostGameAfterModUpdate(MapBean map) {
+    List<ModVersion> selectedModVersions = modManagerController.getSelectedModVersions();
+
+    modService.updateModVersionsAndActivateThose(selectedModVersions)
+        .exceptionally(throwable -> {
+          log.error("Error when updating selected mods", throwable);
+          notificationService.addImmediateErrorNotification(throwable, "game.create.errorUpdatingMods");
+          return null;
+        })
+        .thenAccept(modVersions -> hostGame(map, getUUIDsFromModVersions(modVersions)));
+  }
+
+  @NotNull
+  private Set<String> getUUIDsFromModVersions(List<ModVersion> modVersions) {
+    return modVersions.stream()
         .map(ModVersion::getUid)
         .collect(Collectors.toSet());
+  }
```

Comment:
I think this might be better just as a thenCompose chain on the completableFutures similar to how it is done in the gameService as right now the hostGameAfter naming convention is a little confusing as there are actually multiple steps here